modules/core/client/directives/page-title.client.directive.js

```diff
@@ -19,13 +19,17 @@
 
       function listener(event, toState) {
         var applicationCoreTitle = 'MEAN.js',
-          separeteBy = ' - ';
+          separator = ' - ',
+          stateTitle = applicationCoreTitle + separator;
+
+        toState.name.split('.').forEach(function(value, index) {
+          stateTitle = stateTitle + value.charAt(0).toUpperCase() + value.slice(1) + separator;
+        });
         if (toState.data && toState.data.pageTitle) {
-          var stateTitle = $interpolate(toState.data.pageTitle)($state.$current.locals.globals);
-          element.html(applicationCoreTitle + separeteBy + stateTitle);
-        } else {
-          element.html(applicationCoreTitle);
+          stateTitle = $interpolate(stateTitle + toState.data.pageTitle + separator)(($state.$current.locals.globals));
```

Comment:
Actually i use a combination. For an article, in order to get the **articleResolve.title** and interpolate it to display it. This part just generate the end of the title from the resolve precise by a developer. 

foreach will generate : "MEAN.js - Article - View"
pageTitle will generate : "My first article"
The combination : "MEAN.js - Article - View - My first article"

_I don't think we have this in "toState", it's what i need the pageTitle var. I tried with the the resolve part of the var toState but without success._ 

I see your point, I had thought about it. But if we don't find another way to get the articleResolve.title, without the pageTitle option in the route config ( {{articleResolve.title}} ). This will imply that we will have two options in the route config. 

One to get the end of the automatic pageTitle -> the title of the resource
One to bypass the automatic Title if it's define -> to bypass the use of the State name.

I abandoned this idea to lighten the config file. 

my only personal edit on this PR, in my project is to add 

```
      scope: {
        kind: '=kind'
      },
```

In order to use the directive in my <head>
but also in my content page with kind="header", i will generate and html title with link, and display it in my page. 

Like this : 

<img width="279" alt="screen shot 2016-12-24 at 10 32 09" src="https://cloud.githubusercontent.com/assets/6069599/21466275/3f3bf76c-c9c4-11e6-9be3-49bcafa577be.png">